includes/polling/os/alliedtelesis.inc.php

```diff
@@ -1,21 +1,39 @@
 <?php
 
-// Allied Telesis have somewhat messy MIBs. It's often hard to work out what is where. :)
-$hardware = snmp_get($device, 'atiswitchProductType.0', '-OsvQU', '+AtiSwitch-MIB');
-$version  = snmp_get($device, 'atiswitchSwVersion.0', '-OsvQU', '+AtiSwitch-MIB');
-$software = snmp_get($device, 'atiswitchSw.0', '-OsvQU', '+AtiSwitch-MIB');
+//Below is the original code for Allied Telesis/Telesyn. However testing on recent hardware (last 5ish years), none of these specific mibs work. 
+
+//$hardware = snmp_get($device, 'atiswitchProductType.0', '-OsvQU', '+AtiSwitch-MIB');
+//$version  = snmp_get($device, 'atiswitchSwVersion.0', '-OsvQU', '+AtiSwitch-MIB');
+//$software = snmp_get($device, 'atiswitchSw.0', '-OsvQU', '+AtiSwitch-MIB');
+
+//Have removed above syntax and tried to make below work.
+
+$hardware = trim(snmp_getnext($device, "rscBoardName", "-OQv", "AT-RESOURCE-MIB"), '"');
```

Comment:
What happens for older hardware, will these return any data? If not, you should do a check for empty variables and fall back to the old snmp_get calls.